### PR TITLE
add types for client, some type fixes

### PR DESCRIPTION
**Pull request by SimonFischer04**

### CODE REVIEW

**Comment by twocolors** [comment]
@richardhopton you write ts , what do you think?

File: index.d.ts [hunk 1]
```diff
@@ -487,12 +509,12 @@ declare module "@2colors/esphome-native-api" {
...
             handle: number,
             value: Uint8Array,
-            response = false
+            response?: boolean
         ): Promise<BluetoothGATTWriteResponse>;
         notifyBluetoothGATTCharacteristicService(
             address: number,
             handle: number
-        ): Promise<BluetoothGATTNotifyResponse>;
+        ): Promise<BluetoothGATTNotifyDataResponse>;
```
Comment (richardhopton) [inline index.d.ts:517]: This is not correct. When subscribing to notifications ESPHome will emit a BluetoothGATTNotifyResponse to confirm it is done, BluetoothGATTNotifyDataResponse is only emitted when ESPHome notifies with the value.

**Review by richardhopton — CHANGES_REQUESTED** [review]
This is great - thanks for the PR. One issue that I see with a comment.

File: index.d.ts (hunk 1 above)
Comment (SimonFischer04) [reply index.d.ts]: sorry missed that because type for BluetoothGATTNotifyResponse was missing. Fixed now

**Review by richardhopton — APPROVED** [review]

**Comment by richardhopton** [comment]
LGTM

**Comment by twocolors** [comment]
Thanks guys 👍

**Comment by SimonFischer04** [comment]
Thanks, please also create an npm release, so I can use the fixed types downstream in the iobroker-adapter.

**Comment by twocolors** [comment]
> Thanks, please also create an npm release, so I can use the fixed types downstream in the iobroker-adapter.

ok

(: use Node-Red

**Pull request merged**